compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -793,6 +793,42 @@ where
         }
     }
 
+    pub fn write_uninit(&mut self, dest: &PlaceTy<'tcx, M::PointerTag>) -> InterpResult<'tcx> {
+        let mplace = match dest.place {
+            Place::Ptr(mplace) => MPlaceTy { mplace, layout: dest.layout },
+            Place::Local { frame, local } => {
+                match M::access_local_mut(self, frame, local)? {
+                    Ok(local) => match dest.layout.abi {
+                        Abi::Scalar(_) => {
+                            *local = LocalValue::Live(Operand::Immediate(Immediate::Scalar(
+                                ScalarMaybeUninit::Uninit,
+                            )));
+                            return Ok(());
+                        }
+                        Abi::ScalarPair(..) => {
+                            *local = LocalValue::Live(Operand::Immediate(Immediate::ScalarPair(
+                                ScalarMaybeUninit::Uninit,
+                                ScalarMaybeUninit::Uninit,
+                            )));
+                            return Ok(());
+                        }
+                        _ => self.force_allocation(dest)?,
```

Comment:
This should just `bug!`, it should never happen.

Reply:
I tried this, and it causes an ICE when const evaluating `core::unicode::UNICODE_VERSION`

Reply:
Hm... now I wonder why I thought this was unreachable. :shrug: 

I am a bit surprised that we cannot do this with our existing primitives (this is *extremely* low-level interpreter stuff you are touching here). But... I guess that is to be expected. I haven't worked on this low level in forever so I hope I am not missing something. But it looks reasonable and I mostly tried to set things up such that the 'obvious' mistakes are caught when building this code or at least when running it...